### DISCUSSION TITLE
[Solved] No sound in bluetooth

### DISCUSSION THREAD

**Discussion by floviri** [MiczFlor/RPi-Jukebox-RFID#1607] [Q&A]
Hello,

I just set up Phoniebox and it works great, except for one thing : I have no sound through bluetooth (it works through the jack though).
I followed this tutorial : 
https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Bluetooth
I'm running Phoniebox 2.3 on Buster on a Pi4.

Here's my /etc/asound.conf :
```
pcm.btspeaker {
    type plug
    slave.pcm {
        type bluealsa
        service "org.bluealsa"
        device "74:5C:4B:57:EE:CD"
        profile "a2dp"
        delay -20000
    }
    hint {
        show on
        description "Medion Partybox"
    }
}
```

This test command is working (I hear the sound in my bluetooth headset) :
`aplay -D btspeaker /usr/share/sounds/alsa/Front_Center.wav`

Here is the modified part of my mpd.conf : 
```
audio_output {
        type            "alsa"
        name            "Default Alsa device"    # change if you like
}

audio_output {
        type            "alsa"
        name            "Alsa btspeaker"
        device          "btspeaker"    # see https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/284#issuecomment-443497319
        mixer_type      "software"      # optional
#	device		"hw:0,0"	# optional
#	mixer_type      "hardware"      # optional
#	mixer_device	"default"	# optional
#	mixer_control	"PCM"		# optional
#	mixer_index	"0"		# optional
}
```

**Comment by passuff** [comment]
I'm struggling with exactly the same.  I think you need to modify the mopidy.conf too if you're using the spotify version. But even if you do it, it doesn't work. I can play the test sound with aplay but I don't get any sound from the player.

**Reply by floviri** [reply]
It's working now !
You are right, the audio output needs to be specified in mopidy.conf instead of mpd.conf, maybe because I'm using the Spotify version.
An other thing I had to do was removing the `delay` and `service` parameters in asound.conf, so now I have :

/etc/asound.conf :
```
pcm.btspeaker {
    type plug
    slave.pcm {
        type bluealsa
        device "74:5C:4B:57:EE:CD"
        profile "a2dp"
    }
    hint {
        show on
        description "Medion Partybox"
    }
}
```

And in mopidy.conf :
```
[audio]
#output = autoaudiosink
output = alsasink device=btspeaker 
```
By enabling one line or another, I can switch between the bluetooth headset and the default output !
Now I just need to find a more convenient way to switch, idealy by playing on the headset if it's on, else on the default output…
I'll probably open a new thread, I consider this one solved !

**Reply by passuff** [reply]
I still don't get it working. I tried a fresh install without spotify and it is working. If I use the spotify version it's not even though I configured the mopidy.conf file:
This is my configuration:



/etc/mopidy/mopidy.conf


```
[audio]
output = alsasink
device = btspeaker
mixer_volume = 30

```

 /etc/mpd.conf


```
audio_output {
        type            "alsa"
        name            "My ALSA Device"
        device          "btspeaker"     # optional
        mixer_type      "software"      # optional
#       mixer_device    "default"       # optional
        mixer_control   "Master"        # optional
#       mixer_index     "0"             # optional
}
```


/etc/asound.conf 

```
pcm.btspeaker {
    type plug
    slave.pcm {
        type bluealsa
        #service "org.bluealsa"
        device "0C:A6:94:AF:0E:23"
        profile "a2dp"
        #delay -20000
    }
    hint {
        show on
        description "Medion Partybox"
    }
}
```

Any hint?
